tests/tests_full_scan_engine_udf.py

```diff
@@ -559,13 +560,20 @@ async def test_udf_postgresql_and_mysql(self):
         # Run the query on the aidb database
         logger.info(f'Running query {aidb_query} in aidb database')
         aidb_res = aidb_engine.execute(aidb_query)
-        # # TODO: may have problem with decimal number
         assert len(gt_res) == len(aidb_res)
-        # assert Counter(gt_res) == Counter(aidb_res)
+
+        gt_res = sorted(gt_res)
+        aidb_res = sorted(aidb_res)
+
+        for i in range(len(gt_res)):
+          for element1, element2 in zip(gt_res[i], aidb_res[i]):
+            if isinstance(element1, (int, float, Decimal)):
+              assert abs(int(element1) - int(element2)) <= 1
```

Comment:
it's better to use a relative difference, such as a threshold less than 1%.

Reply:
This will give too much scope for the difference between results for the six decimal point precision it has.. doesn't it? 
Also, for existing tests, this absolute difference works. What do you think?

Reply:
For current tests, '<1' works. However, However, if we conduct additional tests where all values are less than 1, '<1' doesn't make sense. If the result is precise to six decimal places, could you consider using a smaller threshold for accuracy, such as 0.001%? And also, it's better to add a log to print the average difference of all results.